flex/utils.py

```diff
@@ -199,3 +199,14 @@ def dereference_reference(reference, context):
             MESSAGES['reference']['unsupported'].format(reference),
         )
     return jsonpointer.resolve_pointer(context, parts.fragment)
+
+
+def exactly_one(iterable):
+    """
+    Returns True if bool(x) is True for exactly one x in the iterable
+    If the iterable is empty, returns False
+    """
+    it = iter(iterable)
+    # check that one is True, then check that none is True in rest of the
+    # iterator
+    return any(it) and not any(it)
```

Comment:
@maroux @pipermerriam can you advise me on the missing tests for this?

Reply:
This should be tested that for any iterable passed into it:

- returns `True` if **exactly** one item from the iterable evaluates *truthy*
- returns `False` if:
  - None of the elements evaluate *truthy*
  - More than one element evaluates *truthy*
  - The iterable is empty.